### PR TITLE
Add no-prefix option to interface-name

### PR DESCRIPTION
**Pull request by vvakame**
https://github.com/palantir/tslint/issues/512

My implementation allows `IDB` prefix.

```
$ ag "interface I" lib/lib.es6.d.ts
374:interface IArguments {
564:interface IteratorResult<T> {
569:interface Iterator<T> {
575:interface Iterable<T> {
579:interface IterableIterator<T> extends Iterator<T> {
914:interface Int8Array {
931:interface Int8ArrayConstructor {
1017:interface Int16Array {
1038:interface Int16ArrayConstructor {
1087:interface Int32Array {
1104:interface Int32ArrayConstructor {
1604:interface IArguments {
2801:interface Int8Array {
3042:interface Int8ArrayConstructor {
3621:interface Int16Array {
3863:interface Int16ArrayConstructor {
4168:interface Int32Array {
4410:interface Int32ArrayConstructor {
12351:interface IDBCursor {
12375:interface IDBCursorWithValue extends IDBCursor {
12384:interface IDBDatabase extends EventTarget {
12404:interface IDBFactory {
12415:interface IDBIndex {
12433:interface IDBKeyRange {
12449:interface IDBObjectStore {
12472:interface IDBOpenDBRequest extends IDBRequest {
12487:interface IDBRequest extends EventTarget {
12505:interface IDBTransaction extends EventTarget {
12531:interface IDBVersionChangeEvent extends Event {
12541:interface ImageData {
17859:interface IDBEnvironment {
18055:interface IDBObjectStoreParameters {
18060:interface IDBIndexParameters {
18476:interface ITextWriter {
```

IArguments and ITextWriter are maybe unused.

---
:+1: for https://github.com/palantir/tslint/issues/941

### CODE REVIEW

**Comment by JKillian** [comment]
Thanks for the PR @vvakame! Sorry that I've been so delayed in reviewing this, I'll take a look in the next day or two

File: src/rules/interfaceNameRule.ts [hunk 1]
```diff
@@ -45,8 +53,34 @@ class NameWalker extends Lint.RuleWalker {
...
+            return true;
+        }
+
+        const firstCharacter = name.charAt(0);
+        if (firstCharacter !== "I") {
+            return false;
+        }
+
+        const secondCharacter = name.charAt(1);
+        if (!secondCharacter) {
```
Comment (JKillian) [inline src/rules/interfaceNameRule.ts]: We prefer to use explicit comparisons for non-boolean conditions, for example here:
`secondCharacter === ""`

File: src/rules/interfaceNameRule.ts [hunk 2]
```diff
@@ -20,6 +20,7 @@ import * as Lint from "../lint";
 
 export class Rule extends Lint.Rules.AbstractRule {
     public static FAILURE_STRING = "interface name must start with a capitalized I";
+    public static FAILURE_STRING_NO_PREFIX = "interface names must not have a I prefix";
```
Comment (JKillian) [inline src/rules/interfaceNameRule.ts]: "interface name may not have an I prefix"

File: README.md [hunk 3]
```diff
@@ -195,6 +195,7 @@ A sample configuration file with all options is available [here](https://github.
     * `"tabs"` enforces consistent tabs.
     * `"spaces"` enforces consistent spaces.
 * `interface-name` enforces the rule that interface names must begin with a capital 'I'
+    * `"no-prefix"` enforces the rule that interface names must not have a 'I' prefix
```
Comment (JKillian) [inline README.md]: Can we make the options `always-prefix` and `never-prefix`? Sort of following the same pattern as [this PR](https://github.com/palantir/tslint/pull/974/files). If neither option is provided, we'll default to `always-prefix` to maintain backwards compatibility.

**Comment by JKillian** [comment]
Looks good! Left just a few minor changes, if you can make those tweaks before we merge it'd be great. Sorry again for being so slow in reviewing this!

**Comment by vvakame** [comment]
@JKillian thanks for reviewing. I updated this PR. please review again.

**Comment by adidahiya** [comment]
:+1:

**Comment by JKillian** [comment]
Looks good - merged!

**Comment by vvakame** [comment]
:clap:

**Pull request merged**